### PR TITLE
WW-67: Import Excel

### PR DESCRIPTION
**Pull request by TungPhan2808**

### CODE REVIEW

File: src/main/java/com/example/codeE/controller/UserController.java [hunk 1]
```diff
@@ -59,4 +68,11 @@ public ResponseEntity<?> getUser(@PathVariable String userId) {
         User persona = this.userImplement.getUser(userId);
```
Comment (Noro911) [inline src/main/java/com/example/codeE/controller/UserController.java:68]: persona -> user

File: src/main/java/com/example/codeE/mapper/user/UserFromExcel.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class UserFromExcel {
+    @Id
```
Comment (th4tkh13m) [inline src/main/java/com/example/codeE/mapper/user/UserFromExcel.java:18]: Careful of this ID. In the DB, ID is an incremental value, this is maybe undesired.

File: src/main/java/com/example/codeE/service/user/UserImpl.java [hunk 3]
```diff
@@ -40,15 +51,36 @@ public User updateUser(User user) {
...
+        if (ExcelHelper.isValidExcelFile(file)) {
+            try {
+                List<User> users = new ArrayList<>();
+                List<UserFromExcel> importedUsers = ExcelHelper.importFromExcel(file.getInputStream(), UserFromExcel.class);
+                for (UserFromExcel excelUser : importedUsers) {
+                    users.add(new User(excelUser));
+                }
+                for(User user : users){
+                    System.out.println(user.toString());
+                }
```
Comment (th4tkh13m) [inline src/main/java/com/example/codeE/service/user/UserImpl.java:79]: Remove Printing.

File: src/main/java/com/example/codeE/service/user/UserImpl.java [hunk 4]
```diff
@@ -28,6 +35,10 @@ public List<User> getUsersByRoleAndSearchKeyword(GetUsersRequest getUsersRequest
 
     @Override
     public User createUser(User user) {
+        String password = BCryptPassword.generateRandomPassword();
+        user.setUserId(UUID.randomUUID().toString());
+        user.setPassword(password);
+        System.out.println(password);
```
Comment (th4tkh13m) [inline src/main/java/com/example/codeE/service/user/UserImpl.java:41]: Remove this

File: src/main/java/com/example/codeE/security/BCryptPassword.java [hunk 5]
```diff
@@ -1,8 +1,34 @@
...
+        String specialChars = "!@#$%^&*()_+-=[]{}|;:,.<>?";
+        StringBuilder password = new StringBuilder();
+        Random r = new Random();
+        password.append(lower.charAt(r.nextInt(lower.length())));
+        password.append(number.charAt(r.nextInt(number.length())));
+        int remainingLength = 8 - password.length(); // Ensure at least 6 characters
+        String combination = upper + lower + number + specialChars;
+        for (int i = 0; i < remainingLength; i++) {
+            password.append(combination.charAt(r.nextInt(combination.length())));
+        }
```
Comment (th4tkh13m) [inline src/main/java/com/example/codeE/security/BCryptPassword.java:20]: What if the randomized characters don't fall in number/specialChars?

**Review by th4tkh13m — CHANGES_REQUESTED** [review]
Todo:
- Remove printings

**Comment by th4tkh13m** [comment]
Other reviews are not in priorities.

File: src/main/java/com/example/codeE/security/BCryptPassword.java (hunk 5 above)
Comment (TungPhan2808) [reply src/main/java/com/example/codeE/security/BCryptPassword.java:20]: Already handle this in line 14,15

File: src/main/java/com/example/codeE/mapper/user/UserFromExcel.java (hunk 2 above)
Comment (TungPhan2808) [reply src/main/java/com/example/codeE/mapper/user/UserFromExcel.java:18]: An entity class must declare at least 1 '@Id' or '@Embedded' property

File: src/main/java/com/example/codeE/service/user/UserImpl.java (hunk 4 above)
Comment (TungPhan2808) [reply src/main/java/com/example/codeE/service/user/UserImpl.java:41]: why? we need a password for the account

Comment (th4tkh13m) [reply src/main/java/com/example/codeE/service/user/UserImpl.java:41]: Sorry, remove printing =))

**Review by th4tkh13m — APPROVED** [review]

**Pull request merged**